liquidctl/driver/nzxt_smart_device.py

```diff
@@ -182,66 +276,239 @@ def get_status(self, **kwargs):
+            channel_mod = [0x1E, 0x20][cid] # the purpose of this is unknown, but is based on cmd issued by CAM software
+            led_padding = [0xFF, 0xFF, 0xFF]*(maxcolors - color_count)  # set all remaining LEDs to White
+            leds = list(itertools.chain(*colors)) + led_padding
+            self._write([0x22, 0x10, cid+1, 0x00] + leds[0:60]) # send first 20 colors to device (3 bytes per color)
+            self._write([0x22, 0x11, cid+1, 0x00] + leds[60:])  # send remaining colors to device
+            self._write([0x22, 0xA0, cid+1, 0x00, mval, mod3, 0x00, channel_mod, 0x00,
+                         0x00, 0x64, 0x00, 0x32, 0x00, 0x00, 0x01])
+        elif mode == 'wings':  # wings requires special handling
+            for [g, r, b] in colors:
+                self._write([0x22, 0x10, cid+1])  # clear out all independent LEDs
+                self._write([0x22, 0x11, cid+1])  # clear out all independent LEDs
+                color_list = [g, r, b] * 8
+                color_list2 = [int(x // 2.5) for x in color_list]
+                color_list3 = [int(x // 4) for x in color_list2]
+                for i in range(8):   #  send color scheme first, before enabling wings mode
+                    if i == 3 or i == 7:
+                        mod = 0x05
+                    else:
+                        mod = 0x01
+                    msg = ([0x22, 0x20, cid+1, i, 0x04, 0x39, 0x00, mod,
+                            0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x06,
+                            0x05, 0x85, 0x05, 0x85, 0x05, 0x85, 0x00, 0x00,
+                            0x00, 0x00, 0x00, 0x00])
+                    if i == 0 or i == 4:
+                        self._write(msg + color_list)
+                    elif i == 1 or i == 5:
+                        self._write(msg + color_list2)
+                    elif i == 2 or i == 6:
+                        self._write(msg + color_list3)
+                    else:
+                        self._write(msg)
+                self._write([0x22, 0x03, cid+1, 0x08])   # this actually enables wings mode
+        else:
+            channel_mod = [0x01, 0x20][cid]  # the purpose of this is unknown, but is based on cmd issued by CAM software
+            byte7 = mod3
+            byte8 = mod4
+            byte9 = color_count
+            byte10 = 0x00
+            if mval == 0x03:  # for marquee-3|4|5|6 and backwards-marquee-3|4|5|6 we have to put led count in byte9
+                byte9 = mod3
+                byte7 = 0x00
+            elif mval == 0x05:  # for all of the alternating, moving-alternating, and backwards-alternating modes
+                byte7 = 0x00
+                if (mod4 & 0x10) != 0:  # if 'moving' is chosen, set mod3 to 1
+                    byte7 = 0x01
+                byte8 = mod4 & 0x01  # mod4 represents 'backwards' -- need to strip off leading 1s
+                byte10 = mod3   # this is LED count (3 LEDs = 0, 4 LEDs = 1, 5 LEDs = 2, 6 LEDs = 3)
```

Comment:
Suggestion:

```python
byte7 = mod4 & 0x10 >> 4 # moving
byte8 = mod4 & 0x01 # backwards
byte9 = mod3 if mval == 0x03 else color_count # marquee size
byte10 = mod3 if mval == 0x05 else 0x00 # alternating size
```

Not super readable, but a bit easier to follow.

_Please check my assumptions..._

In the general case `byte7 = mod3`, but `mod3` is only non-zero for `mval in [0x03, 0x05]`.  Thus it will be either 0x01 (when `mod4 & 0x10` indicates that's a "moving" variant) or 0x00 in all other cases.

For all modes other than "alternating" `byte8 = mod4 = mod4 & 0x01`, because all other modes don't use the high bits of `mod4`.

Reply:
Those inline `if` `else` statements are very elegant... Yes this works! `byte7` needed parentheses due to precedence issues:
```Python
byte7 = (mod4 & 0x10) >> 4  # sets 'moving' flag for moving alternating modes
byte8 = mod4 & 0x01  # sets 'backwards' flag
byte9 = mod3 if mval == 0x03 else color_count  #  specifies 'marquee' LED size
byte10 = mod3 if mval == 0x05 else 0x00  #  specifies LED size for 'alternating' modes
```
I've checked all the affected color modes -- all okay.